requeJobs & deleteJobs fix

ladislavjacho opened the pull request:
Fix based on new mongodb api:

[https://docs.mongodb.com/manual/reference/method/ObjectId/](https://docs.mongodb.com/manual/reference/method/ObjectId/)

ObjectID -> ObjectId

parialegend commented:
When can we expect this PR to get merge into agenda:master ?

zhangbowy commented:
When can we expect this PR to get merge into agenda:master ?

ladislavjacho commented:
@parialegend  @zhangbowy  I don't know guys. @koresar can help probably. If you need my help, let me know.

imsergiobernal commented:
17 days for just merging, guys...

bmestanov commented on lib/controllers/agendash.js:
```diff
@@ -309,7 +310,8 @@ module.exports = function (agenda, options) {
 
     const collection = agenda._collection.collection || agenda._collection;
```
@ladislavjacho can we remove this declaration, and the commented line below? looks like this PR is not getting merged until tests pass. thanks for the PR, frustrating that this core functionality does not work :/

deyanskiq commented:
Get you merged this PR?

ladislavjacho commented:
@bmestanov  , @willyp63 , @simllll  can you guys proceed please?

mourraille commented:
@bmestanov , @willyp63 , @simllll 
Also waiting for the merge here :)

bmestanov commented:
I'm not a maintainer, guys. Believe me, I wish I could merge too 😕

simllll commented:
@koresar can you do a new release after that pr? Thanks

simllll commented:
@bmestanov can you look into why tests are failing with node 16?

ladislavjacho commented:
> @ladislavjacho can you look into why tests are failing with node 16? 

Upgrading node version to 16 because of failing tests is another topic. I don't recommend it just for merging this tiny boy.

simllll commented:
> > @ladislavjacho can you look into why tests are failing with node 16?
> 
> Upgrading node version to 16 because of failing tests is another topic. I don't recommend it just for merging this tiny boy.

Can you rephrase your answer, I do not understand. The thing is that I'm unable to merge this if the tests are failing.

theharush commented:
this is still broken, any update on why it not getting pushed?

simllll commented:
For me the reason is still node 16 though, or did the master fail with node 16 too before (which i somehow cannot believe?)

theharush commented:
> For me the reason is still node 16 though, or did the master fail with node 16 too before (which i somehow cannot believe?)

i'm running on node 14 on this particular project, and i don't believe it's got anything to do with node, but with mongo.
i believe they change pkFactory behaviour, and this PR suggests a working fix for me. error on my project is pkFactory is not a function. which is true.

debanshu commented:
Hi all, any plans for this merge to happen or is any alternative fix already pushed into master?

ldikmans commented:
what is needed to fix this? Should it run on node16?

simllll commented:
Yes if passes the tests I can merge it immediately

Shahor commented:
Isn't the CI already broken on the main repository already, regardless of this? 
Not saying that it should be fine merging failing stuff but surely we can push this forward? It's breaking agendash for everyone having upgraded and there's no way around it (unless going back in versions).

@ladislavjacho Is there anything I can do to help fix this?

koresar commented:
Hello fellow engineers.

Great news! I fixed the `master` branch unit tests using this very unique, extraordinary rare, and convoluted code:

```
npm i @hapi/hapi@latest
```

I know this looks like magic. My wife always tell me that I'm a magician.

Looks like there was a breaking change is SemVer minor. But, knowing Hapi I think I know how this happens. You can't be a mega-ultra-duper framework and be 100% SemVer compliant each release.

However, you might be surprised too, that this PR breaks mongodb v3.4 compatibility. So, I dropped v3.4 support too. Yay! :) 
Now we have `master` branch which passes all the test! (Awesome me, right?) :)

Also, this means that the next release of Agendash will be SemVer MAJOR - 3.0.0.
Sorry, if this would involve too much work for you people. But WE ARE IN THIS TOGETHER.

Here is the lengthy upgrade instructions:

Step 1.

Open your terminal.

Step 2.

Navigate to the project which is using the awesome `agendash` module:

```
cd ~/i-hate-my-job/duck-processing
```

Step 3.

Run this series of commands:

```
npm i agendash@latest
```

Step 4. [OPTIONAL]

If you are an experienced engineer you might have unit tests in your code. So, likely you would have to run this command too (or not, I'm not your ~~mother~~ manager).

```
npm t
```

Have a good life.

Vasyl

koresar commented:
Published as v3. https://www.npmjs.com/package/agendash/v/3.0.0

ladislavjacho commented:
> Hello fellow engineers.
> 
> Great news! I fixed the `master` branch unit tests using this very unique, extraordinary rare, and convoluted code:
> 
> ```
> npm i @hapi/hapi@latest
> ```
> 
> I know this looks like magic. My wife always tell me that I'm a magician.
> 
> Looks like there was a breaking change is SemVer minor. But, knowing Hapi I think I know how this happens. You can't be a mega-ultra-duper framework and be 100% SemVer compliant each release.
> 
> However, you might be surprised too, that this PR breaks mongodb v3.4 compatibility. So, I dropped v3.4 support too. Yay! :) Now we have `master` branch which passes all the test! (Awesome me, right?) :)
> 
> Also, this means that the next release of Agendash will be SemVer MAJOR - 3.0.0. Sorry, if this would involve too much work for you people. But WE ARE IN THIS TOGETHER.
> 
> Here is the lengthy upgrade instructions:
> 
> Step 1.
> 
> Open your terminal.
> 
> Step 2.
> 
> Navigate to the project which is using the awesome `agendash` module:
> 
> ```
> cd ~/i-hate-my-job/duck-processing
> ```
> 
> Step 3.
> 
> Run this series of commands:
> 
> ```
> npm i agendash@latest
> ```
> 
> Step 4. [OPTIONAL]
> 
> If you are an experienced engineer you might have unit tests in your code. So, likely you would have to run this command too (or not, I'm not your ~mother~ manager).
> 
> ```
> npm t
> ```
> 
> Have a good life.
> 
> Vasyl

Thank you Vasyl! 

I didn't manage to find time for digging deeper into failing tests and happi issues (not my cup of tea).

Dropping support for mongodb 3.4 should not be such a big deal for this project.

Happy that you merged!
